cmd/download.go

```diff
@@ -188,6 +188,9 @@ func newDownload(flags *pflag.FlagSet, usrCfg *viper.Viper) (*download, error) {
 	if err := json.NewDecoder(res.Body).Decode(&d.payload); err != nil {
 		return nil, decodedAPIError(res)
 	}
+	if d.payload.Error.Message != "" {
+		return nil, errors.New(d.payload.Error.Message)
+	}
```

Comment:
On looking at this I'm wondering why the `decodedAPIError` above doesn't return the error. It seems like it should, right?

https://github.com/exercism/cli/blob/4ad01b1562f320ac6c75b9463827b5d5857f4f03/cmd/cmd.go#L98

Reply:
You were right. There wasn't a check for the http status code which was causing the issue